packages/ui/core-components/src/lib/unsorted/viz/table/_DataTable.svelte

```diff
@@ -250,68 +278,78 @@
 	// SORTING
 	// ---------------------------------------------------------------------------------------
 
-	let sortBy = { col: null, ascending: null };
-
-	$: sort = (column) => {
-		if (sortBy.col == column) {
-			sortBy.ascending = !sortBy.ascending;
+	$: sortClick = (column) => {
+		if (sortObj.col == column) {
+			// If the clicked column is the same as inital sort column, switch the sort direction
+			sortObj.ascending = !sortObj.ascending;
 		} else {
-			sortBy.col = column;
-			sortBy.ascending = true;
+			// If the clicked column is different from initial sort column, change the sort column and sort ascending
+			sortObj.col = column;
+			sortObj.ascending = true;
 		}
 
+		sortFunc(sortObj);
+	};
+
+	$: sortFunc = (sortObj) => {
+		const column = sortObj.col;
+
 		// Modifier to sorting function for ascending or descending
-		const sortModifier = sortBy.ascending ? 1 : -1;
+		const sortModifier = sortObj.ascending ? 1 : -1;
 
 		const forceTopOfAscending = (val) =>
 			val === undefined || val === null || (typeof val === 'number' && isNaN(val));
 
-		const sort = (a, b) =>
+		const comparator = (a, b) =>
 			(forceTopOfAscending(a[column]) && !forceTopOfAscending(b[column])) || a[column] < b[column]
 				? -1 * sortModifier
 				: (forceTopOfAscending(b[column]) && !forceTopOfAscending(a[column])) ||
 					  a[column] > b[column]
 					? 1 * sortModifier
 					: 0;
-		data.sort(sort);
-		filteredData = filteredData.sort(sort);
+
+		// data.sort(comparator);
+		const sortedFilteredData = [...filteredData].sort(comparator);
```

Comment:
replaced filteredData = filteredData.sort(sort); to not mutate the original data